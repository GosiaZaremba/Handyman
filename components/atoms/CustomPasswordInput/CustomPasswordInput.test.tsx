import React, {createRef} from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {
  CustomPasswordInput,
  CustomPasswordInputReference,
} from './CustomPasswordInput';

describe('Atom CustomPasswordInput', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(<CustomPasswordInput testID="atom-customPasswordInput" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure reference is passed', () => {
    const inputRef = createRef<CustomPasswordInputReference>();
    const {getByTestId} = render(
      <CustomPasswordInput testID="atom-customPasswordInput" ref={inputRef} />,
    );
    const customPasswordInput = getByTestId('atom-customPasswordInput');
    const customPasswordInputPressable = getByTestId(
      'atom-customPasswordInput-pressable',
    );
    fireEvent.changeText(customPasswordInput, 'input text');
    expect(inputRef.current?.getValue()).toBe('input text');
    expect(inputRef.current?.focus()).toBeUndefined();
    expect(customPasswordInput.props.secureTextEntry).toBe(true);
    expect(customPasswordInputPressable.props.children[0].props.name).toBe(
      'eye-sharp',
    );
  });

  test('ensure password visibility is working', () => {
    const {getByTestId} = render(
      <CustomPasswordInput testID="atom-customPasswordInput" />,
    );
    const customPasswordInputPressable = getByTestId(
      'atom-customPasswordInput-pressable',
    );
    const customPasswordInput = getByTestId('atom-customPasswordInput');

    fireEvent.press(customPasswordInputPressable);
    expect(customPasswordInput.props.secureTextEntry).toBe(false);
    expect(customPasswordInputPressable.props.children[0].props.name).toBe(
      'eye-off-sharp',
    );
    console.log(customPasswordInputPressable.props.children[0]);
  });
});
