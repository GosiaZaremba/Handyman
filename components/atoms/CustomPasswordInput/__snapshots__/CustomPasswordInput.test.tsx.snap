// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom CustomPasswordInput Component Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder=" "
    secureTextEntry={true}
    style={
      Object {
        "fontFamily": "OpenSans-Bold",
        "width": "95%",
      }
    }
    testID="atom-customPasswordInput"
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginRight": 10,
      }
    }
    testID="atom-customPasswordInput-pressable"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#9e9e9e",
            "fontSize": 18,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="atom-customPasswordInput-icon"
    >
      ïŠ¸
    </Text>
  </View>
</View>
`;
